### ISSUE TITLE
CustomField does not support readonly attribute

### ISSUE THREAD

**Issue by mg-dev-fis** [vaadin/flow-components#2953]
Setting readOnly property on component does not affects generated HTML.  I expect that behaviour of component will be similar to TextField or others.

Tested on Vaadin 14.8.4

**Comment by sissbruecker** [comment]
CustomField inherits `setReadOnly` from `AbstractField`, however it's not actually implemented. Whether this should work or not is arguable, as the component would need to "scan" all children, and set them to readonly if possible. We could also just document the behavior, or throw an exception that the operation is not supported. Needs discussion.

Apart from that this should be easy to solve on the application side, by setting the individual child components to read-only.

**Comment by mg-dev-fis** [comment]
Yes, I've fixed it in our application in one line almost but wanted to let you know.

**Comment by sissbruecker** [comment]
We should change the logic for custom field to set a read-only attribute for styling purposes (rather than a property), and document in the JavaDoc that it sets the attribute, but does not toggle the readonly state of the individual inputs.
